### ISSUE TITLE
Can't receive messages

### ISSUE THREAD

**Issue by rtorrero** [davidgfnet/whatsapp-purple#254]
I don't seem to be able to receive any messages. I'm able to see statuses, avatars, and groups, and even I get to see the notification when someone is writing, but the messages never arrive.

The other person reports that the message is marked as "read" for them.

Arch Linux x86_64
Pidgin 2.10.11

Tried both with binary versions of the library as well as compiling it myself.

**Comment by vanfanel** [comment]
I can confirm it happens to me, too.
Sometimes I log in and receive ALL pending messages. Most of the times, I don't.

**Comment by davidgfnet** [comment]
@dx said it had something to do with android encryption... Activate the number again using WART or yowsup. If someone can capture traffic using wireshark and send me a trace (+ password to decrypt it) that would be also nice.

**Comment by estebanvc** [comment]
Same bug here...
Debian Wheezy

**Comment by dequis** [comment]
@davidgfnet misfire

**Comment by Sjd-Risca** [comment]
Yes, I confirm the bug (using Debina Wheezy)

**Comment by estebanvc** [comment]
I have another computer with Debian Wheezy but with a previous version of the plugin. I can receive messages but I get disconnected every X time, when I run pidgin in debug mode I get the message "disconnected from server"

I guess the problem has to do with some mechanism that detects the version of WhatsApp

**Comment by chaydok** [comment]
The same problem here too. I can not receive text but can receive photos.

Fedora 20

**Comment by axel21a** [comment]
I also have problems displaying messages received, not seen, but the warning appears that the contact is writing

**Comment by rtorrero** [comment]
@davidgfnet i'm travelling for a couple of days, when I get home i'll send you the wireshark capture and decrypt password

**Comment by axel21a** [comment]
all resolved with these parameters in the Resource:
Android-2.12.11-443
It had previously S 40-2.12.11-443 and was fine, you see that they have changed something
------------------------------------------------
win 7 --- PRODUCT_VERSION "0.8.1"

**Comment by chaydok** [comment]
@axel21a it not work to me

**Comment by axel21a** [comment]
test  S 40-2.12.11-443

**Comment by axel21a** [comment]
with Android-2.12.11-443 again not seen posts
whereas with S 40-2.12.11-443 seen.
I do not understand anything

**Comment by chaydok** [comment]
With S 40-2.12.11-443,  (19:31:59) Server: unable to parse resource

**Comment by davidgfnet** [comment]
@dequis What do you want me to say? I'm not experiencing this bug in any computer/account. I have more than 4 accounts on linux and windows...
What about you?

**Comment by dequis** [comment]
@davidgfnet by "misfire" i meant you @-mentioned the wrong username, "dx" instead of "dequis". sorry for the confusion :D

Regarding this bug, I haven't heard complaints about anything like this.

**Comment by davidgfnet** [comment]
Oh right :D Sorry for that. Yeah it seems weird to me. I may test it carefully this week.
Thnks!

**Comment by githubtefo** [comment]
I've got the same problem. Every 30-60sec the Server closes the connection.
I tried to change the Resource to Android-2.12.11-443 or S 40-2.12.11-443 with no results.
I use Ubuntu 12.04 with the last dev package and also arch_x86 Pidgin 2.10.11 in both.

**Comment by axel21a** [comment]
right now I work for both resources.
I do not know what about these oddities.
as regards the disconnection, with the version 0.8.1 everything is ok.
use win 7 - italy

**Comment by thomaswpp** [comment]
I have the same problem that gitbubtefo

xubuntu 32bits 14.04

version 0.8.1
Serves: c2.whatsapp.net
Port: 443
Resurce: Android-2.31.151-443
Pidgin 2.10.11 (libpurple 2.10.11)

**Comment by githubtefo** [comment]
Now I Updated to the last arch_x86 purple-whatsapp (0.8.1) and after a few attemps of reconnection Im online and receiving messages (Resurce: Android-2.31.151-443).

**Comment by martijnpc** [comment]
I also have this problem. Occasionally some messages do come in, but it doesn’t last long. Friends seem to receive the messages I send them but I don't get their response.

I am using:
Ubuntu 14.04 64bit on 2 different computers
Pidgin 2.10.9 (libpurple 2.10.9)
Pidgin-whatsapp 0.8.1 (from the Launchpad ppa)
Server: c2.whatsapp.net
Port: 443
Resource:  S 40-2.12.11-443 (I also tried Android-2.31.151-443 and the default)

**Comment by Raaff** [comment]
I can confirm this problem with the binaries from the ppa. Changing the resource from WP to Android or S40 doesn't help. If you compile pidgin-whatsapp yourself, it works. (on ubuntu, first install libfreeimage-dev, then make and sudo make install). That's the reason @davidgfnet is not experiencing this problem, he's using his latest git version. When the binaries are updated, the issue can be marked as fixed.

**Comment by davidgfnet** [comment]
And that kids, is why I always ask the same bloody question: Is this an
issue with git HEAD? People seem to trust any random binary they find on
the internet... Windows legacy I guess..

Thanks for the clarification. And BTW, to the best of my knowledge Fedora
and Ubuntu repos are at 0.8.1 (this is HEAD-1 I think).

**Comment by rybarczykbr** [comment]
Ubuntu PPA version is 0.8.1 and I have the same problem here. I've just forced Synaptic to reinstall the package and then restarted Pidgin. An old message have just arived. Will make some more tests.

**Comment by davidgfnet** [comment]
Good be a good idea to post the sha1 checksum for the .so file (for PPA) so
we can make sure that you guys are using the "good one"

**Comment by salvattore001** [comment]
same problem here. changed resource to S 40-2.12.11-443 and it was fixed

**Comment by davidgfnet** [comment]
Be careful changing resources. I'm pretty sure it's the easiest way to get
your account blocked.

**Comment by salvattore001** [comment]
what do you suggest? i'm using the lastes binary version (0.8.1) for windows

**Comment by martijnpc** [comment]
I compiled the code from the Git repo on one of my machines. I did receive some messages and it seemed to work although I couldn’t test it much.

Here are the sha1 hashes:
````
Git (Commit ef400a7):
sha1sum /usr/lib/purple-2/libwhatsapp.so 
6716abd2d8ced03d879a78f8bda333a7715b6b10  /usr/lib/purple-2/libwhatsapp.so

PPA (V0.8.1):
sha1sum /usr/lib/purple-2/libwhatsapp.so 
48991456a4ab2dbc4981c7a3ba99fc95b5bfc7eb  /usr/lib/purple-2/libwhatsapp.so
````

**Comment by rybarczykbr** [comment]
This is what I got. LinuxMint Xfce (Ubuntu 14.04) 32 bit. PPA (V0.8.1)

/usr/lib/purple-2 $ sha1sum libwhatsapp.so 
ddd2b3a7928c9d3de35c0c3817052d383e178bf6  libwhatsapp.so

martijnpc, are you using 32 or 64 bit?

**Comment by martijnpc** [comment]
@rybarczykbr I use Ubuntu 14.04 64bit.

**Comment by rybarczykbr** [comment]
I've compiled from source and got this SHA
sha1sum libwhatsapp.so 
841f11a58653201a2683b588e29c7ec93906f868  libwhatsapp.so

So I can confirm that the 0.8.1 32bit PPA version is different from github version.

**Comment by rybarczykbr** [comment]
Sadly the problem also happens with the github version.

**Comment by salvattore001** [comment]
now stopped working again

**Comment by davidgfnet** [comment]
Yeah, happens randomly (or appears to be random). I forced it to connect to the same server always and 3 out of 4 connections resulted in messages not being received... I messaged myself to test this. I'll investigate it, but seems pretty random to me...

**Comment by rybarczykbr** [comment]
I've just tried to connect using an empty .purple directory. Got the same problem. My resource was changed to Android (was WP7 before).

The bug seems to start at random when I receive other user "typing notification", that never fades away.

**Comment by davidgfnet** [comment]
Just updated the repo, check with latest version.
It works for me (tested it many times) and I've got one confirmation from a windows user that it fixes the problem. I will close this issue in a day or two if no one complains.

**Comment by rybarczykbr** [comment]
Got 0.8.2 update from PPA. Seems to be working fine. Thanks!

**Comment by pabloscasas** [comment]
In my case with 0.8.1 and 0.8.2 pidgin doesn't even start. I got this from pidgin --debug:
...
(19:36:17) dns: Got response for 'c2.whatsapp.net'
(19:36:17) dnsquery: IP resolved for c2.whatsapp.net
(19:36:17) proxy: Attempting connection to 173.192.222.170
(19:36:17) proxy: Connecting to c2.whatsapp.net:443 with no proxy
(19:36:17) proxy: Connection in progress
Pidgin 2.10.10 ha tenido un fallo y ha intentado generar un archivo «core».
Esto es una errata en el programa y no se ha producido por
algo que haya hecho.

Si puede reproducir este fallo, por favor, notifíquelo a los 
desarrolladores de Pidgin enviando un informe de errata a
http://developer.pidgin.im/simpleticket/

Asegúrese de especificar lo que estaba haciendo en el momento de
suceder esto y envíenos su traza del archivo «core». Si no sabe como
obtener una traza consulte las instrucciones disponibles en
http://developer.pidgin.im/wiki/GetABacktrace
dns[16274]: Oops, father has gone, wait for me, wait...!
Abortado

Up to version 0.8 included, pidgin starts normally. I use gentoo x86_64.

**Comment by davidgfnet** [comment]
Yeah but thats another issue, sorry, closing this one!
Since no one complained I assume it works. They are really quick at complaining not so much at thanking :D

**Comment by rtorrero** [comment]
@davidgfnet I arrived home yesterday and after some testing I can confirm eveything works fine

**Comment by axel21a** [comment]
@davidgfnet everything works fine, thank you

**Comment by githubtefo** [comment]
Suddenly it starts to work fine, with no changes at all (Pidgin 2.10.11, arch_x86 purple-whatsapp (0.8.1) Resource: Android-2.31.151-443).

**Comment by caccamo88** [comment]
to me even with 0.8.2 didn't receive messages from SOME contacts.
Now seems solved with Resource S 40-2.12.11-443
